backend/src/layer/data/flickr_api/api_request.rs

```diff
@@ -0,0 +1,215 @@
+    /// This url is used to create an request with the [`request_license`] function.
+    /// # Errors
+    /// If the request could not be decoded ([`ImageHosterError::DecodeFailed`],
+    /// another request with [`request_err`] will be attempted.
+    /// This error request returns a more detailed error information.
+    /// # Returns
+    /// An Error (as above mentioned).
+    /// True if the given image is hosted under a valid license (see [`json_parser::get_valid_licences`] for more info).
+    /// False if not.
+    pub async fn flickr_photos_licenses_get_license_history(
+        &self,
+        photo_id: &str,
+    ) -> Result<bool, ImageHosterError> {
+        let url = &format!(
+            "{BASE_URL}{GET_LICENCE_HISTORY}{TAG_API_KEY}{api_key}{TAG_PHOTO_ID}{photo_id}{JSON}",
+            BASE_URL = Self::BASE_URL,
+            GET_LICENCE_HISTORY = Self::GET_LICENCE_HISTORY,
+            TAG_API_KEY = Self::TAG_API_KEY,
+            api_key = self.api_key,
+            TAG_PHOTO_ID = Self::TAG_PHOTO_ID,
+            JSON = Self::FORMAT
+        );
+        println!("flickr.photos.licenses.getLicenseHistory URL: {}", url); // TODO remove
+        match self.request_license(&url).await {
+            Ok(licenses) => Ok(self.parser.check_license(licenses)),
+            Err(e) => Err(self.determine_error(url, e).await)
+        }
+    }
+
+    /// This method creates an api request url.
+    /// This url is used to create an request with the [`request_err`] function.
+    /// # Returns
+    /// The hoster error as an [`ImageHosterError`].
+    /// To see all possible cases look here: [`json_parser::parse_error`].
+    async fn determine_error(&self, url: &String, e: ImageHosterError) -> ImageHosterError {
+        if e == ImageHosterError::DecodeFailed {
+            match self.request_err(&url).await {
+                Ok(json_error) => self.parser.parse_error(json_error),
+                Err(e) => e
+            }
+        } else {
+            e
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::interface::image_hoster::ImageHosterError;
+    use crate::layer::data::flickr_api::api_request::ApiRequest;
+    use crate::layer::data::flickr_api::json_parser::JSONParser;
+    use crate::layer::data::flickr_api::json_structs::*;
+    use crate::layer::data::flickr_api::test::const_test_data::{get_api_key, get_licenses_url, get_sizes_url};
+
+    fn get_api_request() -> ApiRequest {
+        ApiRequest {
+            api_key: get_api_key(),
+            parser: JSONParser::new(),
+        }
+    }
+
+    #[tokio::test]
+    async fn valid_request_sizes() {
+        let expected = JsonRootSizes {
+            sizes: Sizes {
+                size: vec![
+                    Size {
+                        label: "Square".to_string(),
+                        width: 75,
+                        height: 75,
+                        source: "https://live.staticflickr.com/65535/52310534489_41350164c9_s.jpg".to_string(),
```

Comment:
it feels a bit dangerous testing against random images, that may not exists in future. are there some dummy images just for such purposes?